src/site/stages/build/plugins/read-assets-from-disk.js

```diff
@@ -0,0 +1,29 @@
+const fs = require('fs');
+const path = require('path');
+
+function readAssetsFromDisk(buildOptions) {
+  const webpackOutput = path.join(
+    __dirname,
+    '../../../../../build/',
+    buildOptions.buildtype,
+    'generated',
+  );
+  return files => {
+    if (fs.existsSync(webpackOutput)) {
+      fs.readdirSync(webpackOutput).forEach(
+        // eslint-disable-next-line no-return-assign
+        filename =>
+          // eslint-disable-next-line no-param-reassign
+          (files[`generated/${filename}`] = {
+            path: `generated/{$filename}`,
+            contents: fs.readFileSync(path.join(webpackOutput, filename)),
+          }),
+      );
+    } else {
+      // eslint-disable-next-line no-console
+      console.log('No Webpack assets found');
+    }
+  };
+}
```

Comment:
So this is like the opposite side of `preserveWebpackOutput`?

Reply:
Not exactly. So previously, when we were running Webpack through Metalsmith, the plugin would take the output and put each file from Webpack into Metalsmith's `files[]`. We then used that for broken link checking and doing the `processDataEntry` thing. Rather than changing that part yet, I decided to just read the Webpack output into `files` like we were doing before.

This _will_ change, but in a future PR.

It might help to think of this as isolating the application build from the content build, but not the content build from the application build. The content build still has a dependency on the application build. The next step in this isolation process is to remove that dependency.